### PR TITLE
Allow add data at repo level

### PR DESCRIPTION
**Pull request by tisonkun**
Ideas: Using GraphQL may speed up the script by reducing a lot of bytes transferred that are unused.

### CODE REVIEW

File: src/build/github.rs [hunk 1]
```diff
@@ -60,14 +61,18 @@ impl Collector {
...
         debug!("collecting contributions");
 
         // Fetch organizations' repositories
         let mut repositories = vec![];
-        for org in orgs {
+        for org in settings.organizations.iter() {
             repositories.extend(self.list_repositories(org).await?);
         }
+        for repo in settings.repositories.iter() {
+            let pair = repo.splitn(2, '/').collect::<Vec<&str>>();
```
Comment (tegioz) [inline src/build/github.rs:73]: Could you validate that the format used in the repositories provided is what we expect (`owner/repo`)?

(i.e. the tool will crash if something like `repo` is provided, or will produce some errors if the format used is `https://github.com/owner/repo`).

We could raise a warning when an invalid entry is detected displaying the correct format.

File: settings/opendal.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Apache OpenDAL settings file
```
Comment (tegioz) [inline settings/opendal.yml:1]: This file shouldn't probably be part of the `contribcard` repository. We left the Kubernetes one as an example until we document it and move it to `docs/config` (similarly to [what we do in the landscape2 repo](https://github.com/cncf/landscape2/tree/main/docs/config)). But projects settings files should probably be hosted somewhere else (actually, Kubernetes' one is hosted at [cncf/contribcard-sites](https://github.com/cncf/contribcard-sites), along with the workflows that build and deploy the site).

**Review by tegioz — CHANGES_REQUESTED** [review]
Thanks @tisonkun 🙂

File: settings/opendal.yml (hunk 2 above)
Comment (tisonkun) [reply settings/opendal.yml:1]: OK. Let me remove this.

Comment (tisonkun) [reply settings/opendal.yml]: Removed.

File: src/build/github.rs (hunk 1 above)
Comment (tisonkun) [reply src/build/github.rs:73]: Updated to best effort verify the input.

File: src/build/github.rs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 //! This module is in charge of collecting contributions from GitHub.
 
 use crate::build::db;
-use anyhow::{bail, Context, Result};
+use anyhow::{bail, ensure, Context, Result};
```
Comment (tegioz) [inline src/build/github.rs:4]: Didn't know about the `ensure` macro, love it! 😍

File: src/build/github.rs [hunk 4]
```diff
@@ -60,14 +61,21 @@ impl Collector {
...
         // Fetch organizations' repositories
         let mut repositories = vec![];
-        for org in orgs {
+        for org in &settings.organizations {
             repositories.extend(self.list_repositories(org).await?);
         }
+        for repo in &settings.repositories {
+            let pair = repo.splitn(2, '/').collect::<Vec<&str>>();
+            ensure!(pair.len() == 2, "repository format must be owner/repo, found: {repo}");
+            ensure!(!pair[0].contains('/'), "owner cannot contain a slash, found: {}", pair[0]);
```
[the remark this answers is not in the document]
Comment (tisonkun) [reply src/build/github.rs:75]: That makes sense. Let me update it.

[the remark this answers is not in the document]
Comment (tisonkun) [reply src/build/github.rs]: Updated.

File: src/build/github.rs (hunk 3 above)
Comment (tisonkun) [reply src/build/github.rs:4]: I learn this one from `snafu` and just noticed that `anyhow` and `error-stack` also have such macros :D

**Review by tegioz — APPROVED** [review]
Thanks @tisonkun!

**Comment by tisonkun** [comment]
Push a new commit to let the formatter happy

**Review by tegioz — APPROVED** [review]

**Pull request merged**